ssl/ssl_lib.c

```diff
@@ -4360,6 +4361,72 @@ const CTLOG_STORE *SSL_CTX_get0_ctlog_store(const SSL_CTX *ctx)
     return ctx->ctlog_store;
 }
 
+void SSL_CTX_set_early_cb(SSL_CTX *c, SSL_early_cb_fn cb, void *arg)
+{
+    c->early_cb = cb;
+    c->early_cb_arg = arg;
+}
+
+int SSL_early_isv2(SSL *s)
+{
+    return s->clienthello->isv2;
+}
+
+unsigned int SSL_early_get0_legacy_version(SSL *s)
+{
+    return s->clienthello->legacy_version;
```

Comment:
Okay, I'll make them return 0 as the least-bad option (unless you want proper success/failure return codes and output parameters, but I don't get that impression).